src/js/components/Diagram/Diagram.js

```diff
@@ -165,75 +162,74 @@ class Diagram extends Component {
-              : 0;
-            const d = COMMANDS[type || 'curved'](
-              points[0],
-              points[1],
-              offsetWidth,
-              anchor,
-            );
-            const strokeWidth = thickness
-              ? parseMetricToNum(theme.global.edgeSize[thickness] || thickness)
-              : 1;
-
-            path = (
-              <path
-                // eslint-disable-next-line react/no-array-index-key
-                key={index}
-                {...cleanedRest}
-                stroke={normalizeColor(
-                  color || theme.diagram.line.color,
-                  theme,
-                )}
-                strokeWidth={strokeWidth}
-                strokeLinecap={round ? 'round' : 'butt'}
-                strokeLinejoin={round ? 'round' : 'miter'}
-                fill="none"
-                d={d}
-              />
-            );
-          }
+    setConnectionPoints(updatedConnectionPoints);
+  };
 
-          return path;
-        },
-      );
+  useEffect(() => {
+    if (!connectionPoints) {
+      placeConnections();
     }
+  }, [connectionPoints]);
+
+  let paths;
+  if (connectionPoints) {
+    paths = connections.map(
+      (
+        { anchor, color, offset, round, thickness, type, ...connectionRest },
+        index,
+      ) => {
+        let path;
+        const cleanedRest = { ...connectionRest };
+        delete cleanedRest.fromTarget;
+        delete cleanedRest.toTarget;
+        const points = connectionPoints[index];
+        if (points) {
+          const offsetWidth = offset
+            ? parseMetricToNum(theme.global.edgeSize[offset])
+            : 0;
+          const d = COMMANDS[type || 'curved'](
+            points[0],
+            points[1],
+            offsetWidth,
+            anchor,
+          );
+          const strokeWidth = thickness
+            ? parseMetricToNum(theme.global.edgeSize[thickness] || thickness)
+            : 1;
+
+          path = (
+            <path
+              // eslint-disable-next-line react/no-array-index-key
+              key={index}
+              {...cleanedRest}
+              stroke={normalizeColor(color || theme.diagram.line.color, theme)}
+              strokeWidth={strokeWidth}
+              strokeLinecap={round ? 'round' : 'butt'}
+              strokeLinejoin={round ? 'round' : 'miter'}
+              fill="none"
+              d={d}
+            />
+          );
+        }
 
-    return (
-      <StyledDiagram
-        ref={this.svgRef}
-        viewBox={`0 0 ${width} ${height}`}
-        preserveAspectRatio="xMinYMin meet"
-        {...rest}
-      >
-        <g>{paths}</g>
-      </StyledDiagram>
+        return path;
+      },
     );
   }
-}
 
+  return (
+    <StyledDiagram
+      ref={svgRef}
+      viewBox={`0 0 ${width} ${height}`}
+      preserveAspectRatio="xMinYMin meet"
+      {...rest}
+    >
+      <g>{paths}</g>
+    </StyledDiagram>
+  );
+};
+
+Diagram.defaultProps = { connections: [] };
```

Comment:
```suggestion
Diagram.defaultProps = {};
```
moving the initialization to the declaration point